repos/system_upgrade/el7toel8/actors/checkremovedpammodules/actor.py

```diff
@@ -0,0 +1,55 @@
+from leapp.actors import Actor
+from leapp.models import Report, PamConfiguration
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.libraries.stdlib import api
+from leapp.libraries.common.reporting import report_with_remediation
+
+
+class CheckRemovedPamModules(Actor):
+    """
+    Check for modules that are not available in RHEL 8 anymore
+
+    At this moment, we check only for pam_tally2. Few more modules
+    are alredy covered in RemoveOldPAMModulesApply actor
+    """
+
+    name = 'removed_pam_modules'
+    consumes = (PamConfiguration, )
+    produces = (Report, )
+    tags = (ChecksPhaseTag, IPUWorkflowTag, )
+
+    def process(self):
+        messages = self.consume(PamConfiguration)
+        config = next(messages)
```

Comment:
you need to handle StopIteration by next(messages, None) and then ofc checking whether you got a message or not. (if not config:...)

otherwise the PR is fine - Sorry for the delay though

Reply:
Seems like this issue is in all the actors that I wrote so far and also in many others. In this case, assuming PAM is mandatory in RHEL, we can require to have exactly one message of this type.

I fixed this one in "fixup" commit to be squashed later. I will have a look into the other occurrences to increase the robustness of the actors.

Reply:
@vinzenz ping?